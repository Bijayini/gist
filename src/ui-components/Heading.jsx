import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import theme from '../lib/theme';

const Heading = ({ text, level, variant, color, className, styles }) => {
  const cssClassName = `level-${level}-${color}-${variant} ${className || ''}`;

  const fontVariant = () => {
    let variantStyles = `font-family: Lato, Sans-Serif;`;
    switch (variant) {
      case 'extra-bold':
        variantStyles += `font-weight: 900`;
        break;
      case 'regular':
        variantStyles += `font-weight: normal`;
        break;
      case 'light':
        variantStyles += `font-weight: 300`;
        break;
      case 'bold':
        variantStyles += `font-weight: 700`;
        break;
      default:
        break;
    }
    return variantStyles;
  };
  return (
    <Fragment>
      {level === '1' && <h1 className={cssClassName}>{text}</h1>}
      {level === '2' && <h2 className={cssClassName}>{text}</h2>}
      {level === '3' && <h3 className={cssClassName}>{text}</h3>}
      <style jsx="true">
        {`
          h1,
          h2,
          h3,
          h4,
          h5 {
            margin: 0;
            padding: 0;
            font-size: 10em;
          }
          .level-1-${color}-${variant} {
            font-size: 1.25rem;
            ${fontVariant()};
            color: ${theme[color]};
          }
          .level-2-${color}-${variant} {
            font-size: 1.125rem;
            ${fontVariant()};
            color: ${theme[color]};
          }
          .level-3-${color}-${variant} {
            font-size: 0.875rem;
            ${fontVariant()};
            color: ${theme[color]};
          }

          @media screen and (${theme.breakpoints.tablet}) {
            .level-1-${color}-${variant} {
              font-size: 1.375rem;
            }
            .level-2-${color}-${variant} {
              font-size: 1.25rem;
            }
            .level-3-${color}-${variant} {
              font-size: 0.875rem;
            }
          }
        `}
      </style>
      {styles}
    </Fragment>
  );
};

Heading.defaultProps = {
  level: '1',
  variant: '',
  color: 'textColor',
  className: '',
  styles: false,
};

Heading.propTypes = {
  text: PropTypes.string.isRequired,
  level: PropTypes.string,
  variant: PropTypes.string,
  color: PropTypes.string,
  className: PropTypes.string,
  styles: PropTypes.node,
};

export default Heading;
