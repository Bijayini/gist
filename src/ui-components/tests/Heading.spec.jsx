import React from 'react';
import { shallow } from 'enzyme';

import Heading from '../Heading';

describe('<Heading />', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      variant: 'extra-bold',
      text: 'Hello',
      level: '1',
    };
  });

  it('should match with snapshot with level 1 and extra-bold variant', () => {
    component = shallow(<Heading {...props} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('should match with snapshot with level 2 and regular variant', () => {
    props.level = '2';
    props.variant = 'regular';
    component = shallow(<Heading {...props} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('should match with snapshot with level 3 and light variant', () => {
    props.level = '3';
    props.variant = 'light';
    component = shallow(<Heading {...props} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('should match with snapshot with level 3 and bold variant', () => {
    props.level = '3';
    props.variant = 'bold';
    component = shallow(<Heading {...props} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('should match with snapshot with level 3 and default variant', () => {
    props.level = '3';
    props.variant = '';
    component = shallow(<Heading {...props} />);
    expect(component.debug()).toMatchSnapshot();
  });
});
