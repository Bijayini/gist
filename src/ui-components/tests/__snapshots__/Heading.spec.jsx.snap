// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match with snapshot with level 1 and extra-bold variant 1`] = `
"<Fragment>
  <h1 className=\\"jsx-2198291737 level-1-textColor-extra-bold \\">
    Hello
  </h1>
  <JSXStyle id=\\"1267284431\\" dynamic={{...}}>
    h1.__jsx-style-dynamic-selector,h2.__jsx-style-dynamic-selector,h3.__jsx-style-dynamic-selector,h4.__jsx-style-dynamic-selector,h5.__jsx-style-dynamic-selector{margin:0;padding:0;font-size:10em;}.level-1-textColor-extra-bold.__jsx-style-dynamic-selector{font-size:1.25rem;font-family: Lato, Sans-Serif;font-weight: 900;color:#4a4a4a;}.level-2-textColor-extra-bold.__jsx-style-dynamic-selector{font-size:1.125rem;font-family: Lato, Sans-Serif;font-weight: 900;color:#4a4a4a;}.level-3-textColor-extra-bold.__jsx-style-dynamic-selector{font-size:0.875rem;font-family: Lato, Sans-Serif;font-weight: 900;color:#4a4a4a;}@media screen and (min-width: 768px){.level-1-textColor-extra-bold.__jsx-style-dynamic-selector{font-size:1.375rem;}.level-2-textColor-extra-bold.__jsx-style-dynamic-selector{font-size:1.25rem;}.level-3-textColor-extra-bold.__jsx-style-dynamic-selector{font-size:0.875rem;}}
  </JSXStyle>
</Fragment>"
`;

exports[`<Heading /> should match with snapshot with level 2 and regular variant 1`] = `
"<Fragment>
  <h2 className=\\"jsx-2389411635 level-2-textColor-regular \\">
    Hello
  </h2>
  <JSXStyle id=\\"1267284431\\" dynamic={{...}}>
    h1.__jsx-style-dynamic-selector,h2.__jsx-style-dynamic-selector,h3.__jsx-style-dynamic-selector,h4.__jsx-style-dynamic-selector,h5.__jsx-style-dynamic-selector{margin:0;padding:0;font-size:10em;}.level-1-textColor-regular.__jsx-style-dynamic-selector{font-size:1.25rem;font-family: Lato, Sans-Serif;font-weight: normal;color:#4a4a4a;}.level-2-textColor-regular.__jsx-style-dynamic-selector{font-size:1.125rem;font-family: Lato, Sans-Serif;font-weight: normal;color:#4a4a4a;}.level-3-textColor-regular.__jsx-style-dynamic-selector{font-size:0.875rem;font-family: Lato, Sans-Serif;font-weight: normal;color:#4a4a4a;}@media screen and (min-width: 768px){.level-1-textColor-regular.__jsx-style-dynamic-selector{font-size:1.375rem;}.level-2-textColor-regular.__jsx-style-dynamic-selector{font-size:1.25rem;}.level-3-textColor-regular.__jsx-style-dynamic-selector{font-size:0.875rem;}}
  </JSXStyle>
</Fragment>"
`;

exports[`<Heading /> should match with snapshot with level 3 and bold variant 1`] = `
"<Fragment>
  <h3 className=\\"jsx-1872306711 level-3-textColor-bold \\">
    Hello
  </h3>
  <JSXStyle id=\\"1267284431\\" dynamic={{...}}>
    h1.__jsx-style-dynamic-selector,h2.__jsx-style-dynamic-selector,h3.__jsx-style-dynamic-selector,h4.__jsx-style-dynamic-selector,h5.__jsx-style-dynamic-selector{margin:0;padding:0;font-size:10em;}.level-1-textColor-bold.__jsx-style-dynamic-selector{font-size:1.25rem;font-family: Lato, Sans-Serif;font-weight: 700;color:#4a4a4a;}.level-2-textColor-bold.__jsx-style-dynamic-selector{font-size:1.125rem;font-family: Lato, Sans-Serif;font-weight: 700;color:#4a4a4a;}.level-3-textColor-bold.__jsx-style-dynamic-selector{font-size:0.875rem;font-family: Lato, Sans-Serif;font-weight: 700;color:#4a4a4a;}@media screen and (min-width: 768px){.level-1-textColor-bold.__jsx-style-dynamic-selector{font-size:1.375rem;}.level-2-textColor-bold.__jsx-style-dynamic-selector{font-size:1.25rem;}.level-3-textColor-bold.__jsx-style-dynamic-selector{font-size:0.875rem;}}
  </JSXStyle>
</Fragment>"
`;

exports[`<Heading /> should match with snapshot with level 3 and default variant 1`] = `
"<Fragment>
  <h3 className=\\"jsx-3253502148 level-3-textColor- \\">
    Hello
  </h3>
  <JSXStyle id=\\"1267284431\\" dynamic={{...}}>
    h1.__jsx-style-dynamic-selector,h2.__jsx-style-dynamic-selector,h3.__jsx-style-dynamic-selector,h4.__jsx-style-dynamic-selector,h5.__jsx-style-dynamic-selector{margin:0;padding:0;font-size:10em;}.level-1-textColor-.__jsx-style-dynamic-selector{font-size:1.25rem;font-family: Lato, Sans-Serif;;color:#4a4a4a;}.level-2-textColor-.__jsx-style-dynamic-selector{font-size:1.125rem;font-family: Lato, Sans-Serif;;color:#4a4a4a;}.level-3-textColor-.__jsx-style-dynamic-selector{font-size:0.875rem;font-family: Lato, Sans-Serif;;color:#4a4a4a;}@media screen and (min-width: 768px){.level-1-textColor-.__jsx-style-dynamic-selector{font-size:1.375rem;}.level-2-textColor-.__jsx-style-dynamic-selector{font-size:1.25rem;}.level-3-textColor-.__jsx-style-dynamic-selector{font-size:0.875rem;}}
  </JSXStyle>
</Fragment>"
`;

exports[`<Heading /> should match with snapshot with level 3 and light variant 1`] = `
"<Fragment>
  <h3 className=\\"jsx-4063421011 level-3-textColor-light \\">
    Hello
  </h3>
  <JSXStyle id=\\"1267284431\\" dynamic={{...}}>
    h1.__jsx-style-dynamic-selector,h2.__jsx-style-dynamic-selector,h3.__jsx-style-dynamic-selector,h4.__jsx-style-dynamic-selector,h5.__jsx-style-dynamic-selector{margin:0;padding:0;font-size:10em;}.level-1-textColor-light.__jsx-style-dynamic-selector{font-size:1.25rem;font-family: Lato, Sans-Serif;font-weight: 300;color:#4a4a4a;}.level-2-textColor-light.__jsx-style-dynamic-selector{font-size:1.125rem;font-family: Lato, Sans-Serif;font-weight: 300;color:#4a4a4a;}.level-3-textColor-light.__jsx-style-dynamic-selector{font-size:0.875rem;font-family: Lato, Sans-Serif;font-weight: 300;color:#4a4a4a;}@media screen and (min-width: 768px){.level-1-textColor-light.__jsx-style-dynamic-selector{font-size:1.375rem;}.level-2-textColor-light.__jsx-style-dynamic-selector{font-size:1.25rem;}.level-3-textColor-light.__jsx-style-dynamic-selector{font-size:0.875rem;}}
  </JSXStyle>
</Fragment>"
`;
